code/game/objects/random/random.dm

```diff
@@ -278,3 +276,234 @@
 					/obj/item/toy/plushie/kitten,\
 					/obj/item/toy/plushie/lizard)
 
+/obj/random/glowstick
+	name = "random glowstick"
+	desc = "This is a random glowstick."
+	icon = 'icons/obj/glowsticks.dmi'
+	icon_state = "glowstick"
+	item_to_spawn()
+		return pick(/obj/item/device/flashlight/glowstick,\
+					/obj/item/device/flashlight/glowstick/red,\
+					/obj/item/device/flashlight/glowstick/blue,\
+					/obj/item/device/flashlight/glowstick/orange,\
+					/obj/item/device/flashlight/glowstick/yellow)
+
+/obj/random/booze
+	name = "random alcoholic drink"
+	desc = "This is a random alcoholic drink."
+	icon = 'icons/obj/drinks.dmi'
+	icon_state = "broken_bottle"
+	item_to_spawn()
+		return pick(/obj/item/weapon/reagent_containers/food/drinks/bottle/gin,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/whiskey,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/vodka,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/tequilla,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/bottleofnothing,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/rum,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/vermouth,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/kahlua,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/cognac,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/wine,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/absinthe,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/melonliquor,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/pwine,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/brandy,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/guinnes,\
+					/obj/item/weapon/reagent_containers/food/drinks/bottle/drambuie,\
+					/obj/item/weapon/reagent_containers/food/drinks/cans/beer,\
+					/obj/item/weapon/reagent_containers/food/drinks/cans/ale)
+
+/obj/random/melee
+	name = "random melee weapon"
+	desc = "This is a random melee weapon."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "baton"
+	item_to_spawn()
+		return pick(/obj/item/weapon/melee/telebaton,\
+					/obj/item/weapon/melee/energy/sword,\
+					/obj/item/weapon/melee/energy/sword/pirate,\
+					/obj/item/weapon/melee/energy/glaive,\
+					/obj/item/weapon/melee/chainsword,\
+					/obj/item/weapon/melee/baton/stunrod,\
+					/obj/item/weapon/material/harpoon,\
+					/obj/item/weapon/material/scythe,\
+					/obj/item/weapon/material/twohanded/spear/plasteel,\
+					/obj/item/weapon/material/sword/trench,\
+					/obj/item/weapon/material/sword/rapier)
+
+/obj/random/coin
+	name = "Random Coin"
+	desc = "This is a random coin."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "coin"
+	item_to_spawn()
+		return pick(prob(5);/obj/item/weapon/coin/iron,\
+					prob(3);/obj/item/weapon/coin/silver,\
```

Comment:
Still has probabilities attached.

Meaning a probability is generated for every item on this list, every time a random coin is spawned.